README.md
# simple_design

A simple design library for Flutter.

This package includes:
- Premade themes, dark and light brightness
- Several custom Simple Design widgets, serving the original design idea.

Link to the example images: https://imgur.com/a/duatQYT

For general design advice I used the Material Design Specifications from material.io.

## Font

I recommend using the free-for-commercial font "HK Grotesk", which is used in the example
and also defined as the standart font in the themes.

But really and sans-serif, geometric or grotesk typeface should work well (for example 
Lineto Circular).

## SDAppBar and SDSliverAppBar

These two widgets are meant to be used instead of the regular AppBar/SliverAppBar.
They use certain styling that otherwise would have to be applied to every AppBar in
your app.

They are customizable with title, leading, actions, bottom, automaticallyImplyLeading
and flexibeSpace. The other values are not customizable to not negatively impact the
appearance of your app.

![SDAppBar and SDSliverAppBar](https://i.imgur.com/Mh7btbo.jpg)

## SDDialog with SDDismissButton and SDActionButton

This is a custom dialog and should be used in the showDialog() method instead of
AlertDialog or SimpleDialog. It can be given values for title, content, a SDDismissButton
for a dismiss button, a SDActionButton for an action button and a bool barrierDismissable,
which is not yet working.

The SDDismissButton and SDActionButton should be self-explaining.

![SDDialog with SDDismissButton and SDActionButton](https://i.imgur.com/uDETYMG.jpg)

## SDDivider and SDSectionHeader

Designed for the use in Colums and Lists, the SDDivider provides a vertical padding of 36.0
pixels, which is recommended, but can be customized with the height property.

The SDSectionHeader is to be used below a SDDivider to announce the title of a new section,
as it can be seen in the example application.

![SDSectionHeader](https://i.imgur.com/DF9O8Ok.jpg)

## SDCard

The SDCard widget is a highly customizable card, with options for mediaContent, content, title,
subtitle, actions and backgroundColor. It has a slight shadow underneath and is best used in
Colums and Lists.

![SDCard](https://i.imgur.com/ScrMJAH.jpg)

CHANGELOG.md
## [1.0.0] - 16.12.2018.

* Initial Release with Light and Dark Themes and several widgets.

## [1.0.1] - 16.12.2018.

* Minor fixes

## [1.0.2] - 16.12.2018.

* Minor fixes

## [1.0.3] - 16.12.2018.

* Fixed a brightness bug in the AppBar

lib/simple_design.dart
library simple_design;

import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

class SimpleDesign {
  static final lightTheme = ThemeData(
      splashColor: Colors.blueAccent.withOpacity(0.15),
      textTheme: TextTheme(
          title: TextStyle(fontWeight: FontWeight.bold, fontSize: 20.0),
          body1: TextStyle(fontSize: 16.0),
          body2: TextStyle(fontSize: 16.0, color: Colors.grey[800]),
          button: TextStyle(fontSize: 14.0, fontWeight: FontWeight.bold),
          caption:
              TextStyle(fontSize: 16.0, color: Colors.black.withOpacity(0.5))),
      accentColor: Colors.blueAccent,
      accentColorBrightness: Brightness.dark,
      backgroundColor: Colors.grey[200],
      bottomAppBarColor: Colors.white,
      brightness: Brightness.light,
      buttonColor: Colors.blueAccent,
      buttonTheme: ButtonThemeData(
          textTheme: ButtonTextTheme.primary,
          buttonColor: Colors.blueAccent,
          alignedDropdown: true,
          disabledColor: Colors.grey[400],
          height: 44.0,
          highlightColor: Colors.transparent,
          layoutBehavior: ButtonBarLayoutBehavior.constrained,
          shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.circular(4.0),
              side: BorderSide(style: BorderStyle.none))),
      highlightColor: Colors.transparent,
      disabledColor: Colors.grey[400],
      canvasColor: Colors.white,
      cardColor: Colors.white,
      cursorColor: Colors.blueAccent,
      dialogBackgroundColor: Colors.white,
      dialogTheme: DialogTheme(
          shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.circular(4.0),
              side: BorderSide(style: BorderStyle.none))),
      dividerColor: Colors.transparent,
      errorColor: Colors.red[400],
      inputDecorationTheme: InputDecorationTheme(
        border: UnderlineInputBorder(),
        contentPadding: EdgeInsets.all(8.0),
        hasFloatingPlaceholder: false,
        errorMaxLines: 1,
        filled: false,
      ),
      indicatorColor: Colors.blueAccent,
      primaryColor: Colors.blueAccent,
      primaryColorBrightness: Brightness.light,
      primaryColorDark: Colors.blueAccent[400],
      primaryColorLight: Colors.blueAccent[100],
      scaffoldBackgroundColor: Colors.white,
      selectedRowColor: Colors.grey[200],
      sliderTheme: SliderThemeData(
          activeTrackColor: Colors.blueAccent,
          inactiveTrackColor: Colors.grey[400],
          disabledActiveTrackColor: Colors.grey[400],
          disabledInactiveTrackColor: Colors.grey[300],
          activeTickMarkColor: Colors.blueAccent,
          inactiveTickMarkColor: Colors.grey,
          disabledActiveTickMarkColor: Colors.grey[400],
          disabledInactiveTickMarkColor: Colors.grey[400],
          thumbColor: Colors.blueAccent,
          disabledThumbColor: Colors.grey[400],
          overlayColor: Colors.blueAccent.withOpacity(0.3),
          valueIndicatorColor: Colors.blueAccent,
          thumbShape: RoundSliderThumbShape(),
          valueIndicatorShape: PaddleSliderValueIndicatorShape(),
          showValueIndicator: ShowValueIndicator.onlyForDiscrete,
          valueIndicatorTextStyle:
              TextStyle(fontWeight: FontWeight.bold, fontSize: 14.0)),
      fontFamily: "HK Grotesk",
      splashFactory: InkRipple.splashFactory,
      tabBarTheme: TabBarTheme(
          indicator: BoxDecoration(
              border: Border(
                  bottom: BorderSide(
                      color: Colors.blueAccent,
                      width: 2.0,
                      style: BorderStyle.solid)),
              gradient: LinearGradient(
                  begin: Alignment.bottomCenter,
                  end: Alignment(0.0, 0.8),
                  colors: [
                    Colors.blueAccent.withOpacity(0.5),
                    Colors.blueAccent.withOpacity(0.0)
                  ])),
          indicatorSize: TabBarIndicatorSize.label,
          labelColor: Colors.blueAccent,
          unselectedLabelColor: Colors.grey),
      textSelectionColor: Colors.blueAccent.withOpacity(0.5),
      textSelectionHandleColor: Colors.blueAccent,
      toggleableActiveColor: Colors.blueAccent,
      unselectedWidgetColor: Colors.grey[600]);

  static final darkTheme = ThemeData(
      splashColor: Colors.white.withOpacity(0.1),
      textTheme: TextTheme(
          title: TextStyle(fontWeight: FontWeight.bold, fontSize: 20.0),
          body1: TextStyle(fontSize: 16.0),
          body2: TextStyle(fontSize: 16.0, color: Colors.grey[200]),
          button: TextStyle(fontSize: 14.0, fontWeight: FontWeight.bold),
          display1: TextStyle(fontSize: 20.0),
          display2: TextStyle(fontSize: 24.0),
          display3: TextStyle(fontSize: 28.0),
          display4: TextStyle(fontSize: 32.0),
          overline: TextStyle(fontSize: 14.0),
          caption:
              TextStyle(fontSize: 16.0, color: Colors.white.withOpacity(0.5))),
      primaryTextTheme: TextTheme(
          title: TextStyle(
              fontWeight: FontWeight.bold, fontSize: 20.0, color: Colors.white),
          body1: TextStyle(fontSize: 16.0, color: Colors.white),
          body2: TextStyle(fontSize: 16.0, color: Colors.grey[200]),
          button: TextStyle(
              fontSize: 14.0, fontWeight: FontWeight.bold, color: Colors.white),
          display1: TextStyle(fontSize: 20.0, color: Colors.white),
          display2: TextStyle(fontSize: 24.0, color: Colors.white),
          display3: TextStyle(fontSize: 28.0, color: Colors.white),
          display4: TextStyle(fontSize: 32.0, color: Colors.white),
          overline: TextStyle(fontSize: 14.0, color: Colors.white),
          caption:
              TextStyle(fontSize: 16.0, color: Colors.white.withOpacity(0.5))),
      accentColor: Colors.blueAccent,
      accentColorBrightness: Brightness.dark,
      backgroundColor: Colors.grey[800],
      bottomAppBarColor: Colors.grey[850],
      brightness: Brightness.dark,
      primaryIconTheme: IconThemeData(color: Colors.white),
      buttonColor: Colors.blueAccent,
      buttonTheme: ButtonThemeData(
          textTheme: ButtonTextTheme.primary,
          buttonColor: Colors.blueAccent,
          alignedDropdown: true,
          disabledColor: Colors.grey[600],
          height: 44.0,
          highlightColor: Colors.transparent,
          layoutBehavior: ButtonBarLayoutBehavior.constrained,
          shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.circular(4.0),
              side: BorderSide(style: BorderStyle.none))),
      highlightColor: Colors.transparent,
      disabledColor: Colors.grey[600],
      canvasColor: Colors.grey[850],
      cardColor: Colors.grey[850],
      cursorColor: Colors.blueAccent,
      dialogBackgroundColor: Colors.grey[850],
      dialogTheme: DialogTheme(
          shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.circular(4.0),
              side: BorderSide(style: BorderStyle.none))),
      dividerColor: Colors.transparent,
      errorColor: Colors.red[400],
      inputDecorationTheme: InputDecorationTheme(
        border: UnderlineInputBorder(),
        contentPadding: EdgeInsets.all(8.0),
        hasFloatingPlaceholder: false,
        errorMaxLines: 1,
        filled: false,
      ),
      indicatorColor: Colors.blueAccent,
      primaryColor: Colors.blueAccent,
      primaryColorBrightness: Brightness.light,
      primaryColorDark: Colors.blueAccent[400],
      primaryColorLight: Colors.blueAccent[100],
      scaffoldBackgroundColor: Colors.grey[850],
      selectedRowColor: Colors.grey[800],
      sliderTheme: SliderThemeData(
          activeTrackColor: Colors.blueAccent,
          inactiveTrackColor: Colors.grey[400],
          disabledActiveTrackColor: Colors.grey[400],
          disabledInactiveTrackColor: Colors.grey[300],
          activeTickMarkColor: Colors.blueAccent,
          inactiveTickMarkColor: Colors.grey,
          disabledActiveTickMarkColor: Colors.grey[400],
          disabledInactiveTickMarkColor: Colors.grey[400],
          thumbColor: Colors.blueAccent,
          disabledThumbColor: Colors.grey[400],
          overlayColor: Colors.blueAccent.withOpacity(0.3),
          valueIndicatorColor: Colors.blueAccent,
          thumbShape: RoundSliderThumbShape(),
          valueIndicatorShape: PaddleSliderValueIndicatorShape(),
          showValueIndicator: ShowValueIndicator.onlyForDiscrete,
          valueIndicatorTextStyle:
              TextStyle(fontWeight: FontWeight.bold, fontSize: 14.0)),
      splashFactory: InkRipple.splashFactory,
      fontFamily: "HK Grotesk",
      tabBarTheme: TabBarTheme(
          indicator: BoxDecoration(
              border: Border(
                  bottom: BorderSide(
                      color: Colors.blueAccent,
                      width: 2.0,
                      style: BorderStyle.solid)),
              gradient: LinearGradient(
                  begin: Alignment.bottomCenter,
                  end: Alignment(0.0, 0.8),
                  colors: [
                    Colors.blueAccent.withOpacity(0.5),
                    Colors.blueAccent.withOpacity(0.0)
                  ])),
          indicatorSize: TabBarIndicatorSize.label,
          labelColor: Colors.blueAccent,
          unselectedLabelColor: Colors.grey[400]),
      textSelectionColor: Colors.blueAccent.withOpacity(0.5),
      textSelectionHandleColor: Colors.blueAccent,
      toggleableActiveColor: Colors.blueAccent,
      unselectedWidgetColor: Colors.grey[400]);
}

class SDAppBar extends StatelessWidget implements PreferredSizeWidget {
  final Widget title;
  final Widget leading;
  final List<Widget> actions;
  final PreferredSizeWidget bottom;
  final bool automaticallyImplyLeading;
  final Widget flexibleSpace;

  SDAppBar(
      {@required this.title,
      this.leading,
      this.actions,
      this.bottom,
      this.automaticallyImplyLeading: true,
      this.flexibleSpace});

  @override
  Widget build(BuildContext context) {
    return AppBar(
      brightness: Theme.of(context).brightness,
      bottom: bottom,
      centerTitle: true,
      automaticallyImplyLeading: automaticallyImplyLeading,
      flexibleSpace: flexibleSpace,
      leading: leading,
      title: title,
      elevation: 0.0,
      backgroundColor: Theme.of(context).scaffoldBackgroundColor,
      actions: actions,
    );
  }

  @override
  Size get preferredSize => Size(double.infinity, kToolbarHeight);
}

class SDSliverAppBar extends StatelessWidget {
  final Widget title;
  final Widget leading;
  final List<Widget> actions;
  final PreferredSizeWidget bottom;
  final bool automaticallyImplyLeading;
  final Widget flexibleSpace;
  final bool pinned;
  final bool floating;
  final bool snap;
  final double expandedHeight;

  SDSliverAppBar(
      {@required this.title,
      this.leading,
      this.actions,
      this.bottom,
      this.pinned: true,
      this.floating: false,
      this.expandedHeight: 0.0,
      this.snap: false,
      this.automaticallyImplyLeading: true,
      this.flexibleSpace});

  @override
  Widget build(BuildContext context) {
    return SliverAppBar(
      forceElevated: bottom != null,
      brightness: Theme.of(context).brightness,
      bottom: bottom,
      centerTitle: true,
      automaticallyImplyLeading: automaticallyImplyLeading,
      flexibleSpace: flexibleSpace,
      leading: leading,
      title: title,
      elevation: 1.0,
      backgroundColor: Theme.of(context).scaffoldBackgroundColor,
      actions: actions,
      pinned: pinned,
      floating: floating,
      expandedHeight: expandedHeight,
      snap: snap,
    );
  }
}

class SDDialog extends StatelessWidget {
  final String title;
  final Widget content;
  final SDDismissButton dismissButton;
  final SDActionButton actionButton;
  final bool barrierDismissible;

  SDDialog(
      {@required this.title,
      this.barrierDismissible: false,
      this.content,
      this.dismissButton,
      this.actionButton});

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.all(24.0),
      child: Center(
        child: Material(
          elevation: 3.0,
          clipBehavior: Clip.antiAlias,
          borderRadius: BorderRadius.circular(4.0),
          color: Theme.of(context).dialogBackgroundColor,
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: <Widget>[
              Padding(
                padding: const EdgeInsets.all(24.0),
                child: Text(
                  title,
                  style: Theme.of(context).textTheme.title,
                ),
              ),
              Padding(
                padding: const EdgeInsets.fromLTRB(24.0, 0.0, 24.0, 24.0),
                child: content ?? SizedBox(),
              ),
              Container(
                color: Theme.of(context).backgroundColor,
                child: Padding(
                  padding: const EdgeInsets.symmetric(vertical: 12.0),
                  child: ButtonBar(
                    mainAxisSize: MainAxisSize.max,
                    children: <Widget>[
                      dismissButton ?? SizedBox(),
                      actionButton ?? SizedBox()
                    ],
                  ),
                ),
              )
            ],
            mainAxisSize: MainAxisSize.min,
          ),
        ),
      ),
    );
  }
}

class SDDismissButton extends StatelessWidget {
  final VoidCallback onPressed;
  final String title;
  final Icon icon;

  SDDismissButton({@required this.onPressed, this.title: "Cancel", this.icon});

  @override
  Widget build(BuildContext context) {
    return FlatButton(
        onPressed: onPressed,
        textColor: Color.fromRGBO(
            255 - Theme.of(context).backgroundColor.red,
            255 - Theme.of(context).backgroundColor.green,
            255 - Theme.of(context).backgroundColor.blue,
            1.0),
        child: icon == null
            ? Text(title)
            : Row(
                children: <Widget>[icon, SizedBox(width: 4.0), Text(title)],
              ));
  }
}

class SDActionButton extends StatelessWidget {
  final VoidCallback onPressed;
  final String title;
  final Icon icon;
  final Color color;

  SDActionButton(
      {@required this.onPressed, this.title: "Ok", this.icon, this.color});

  @override
  Widget build(BuildContext context) {
    return FlatButton(
        color: color ?? Theme.of(context).primaryColor,
        onPressed: onPressed,
        child: icon == null
            ? Text(title)
            : Row(
                children: <Widget>[icon, SizedBox(width: 4.0), Text(title)],
              ));
  }
}

class SDDivider extends StatelessWidget {
  final double height;

  SDDivider({this.height: 36.0});

  @override
  Widget build(BuildContext context) {
    return SizedBox(height: height);
  }
}

class SDSectionHeader extends StatelessWidget {
  final String title;

  SDSectionHeader(this.title);

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 8.0, horizontal: 16.0),
      child: Text(
        title.toUpperCase(),
        style: Theme.of(context).textTheme.caption,
      ),
    );
  }
}

class SDCard extends StatelessWidget {
  final Widget mediaContent;
  final Widget content;
  final String title;
  final String subtitle;
  final List<Widget> actions;
  final Color backgroundColor;

  SDCard(
      {this.mediaContent,
      @required this.content,
      this.subtitle,
      this.title,
      this.actions,
      this.backgroundColor});

  @override
  Widget build(BuildContext context) {
    return Material(
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        mainAxisSize: MainAxisSize.min,
        children: <Widget>[
          LimitedBox(
              maxHeight: 250.0,
              child: Row(
                children: <Widget>[
                  Expanded(child: mediaContent ?? SizedBox()),
                ],
              )),
          title != null
              ? Padding(
                  padding: const EdgeInsets.fromLTRB(20.0, 20.0, 20.0, 0.0),
                  child: Text(
                    title,
                    style: Theme.of(context).textTheme.title,
                  ),
                )
              : SizedBox(),
          subtitle != null
              ? Padding(
                  padding: const EdgeInsets.fromLTRB(20.0, 0.0, 20.0, 0.0),
                  child: Text(
                    subtitle,
                    style: Theme.of(context).textTheme.subtitle,
                  ),
                )
              : SizedBox(),
          Padding(
            padding: const EdgeInsets.fromLTRB(20.0, 16.0, 20.0, 20.0),
            child: content,
          ),
          actions != null
              ? Material(
                  color: Theme.of(context).backgroundColor,
                  child: Padding(
                    padding: const EdgeInsets.symmetric(vertical: 12.0),
                    child: ButtonBar(
                      mainAxisSize: MainAxisSize.max,
                      children: actions,
                    ),
                  ),
                )
              : SizedBox()
        ],
      ),
      color: backgroundColor ?? Theme.of(context).cardColor,
      borderRadius: BorderRadius.circular(4.0),
      clipBehavior: Clip.antiAlias,
      elevation: 1.0,
      shadowColor: Colors.black.withOpacity(0.5),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:simple_design/simple_design.dart';

final ThemeData theme = SimpleDesign.lightTheme;

void main() {
  runApp(MyApp());
  SystemChrome.setSystemUIOverlayStyle(
    SystemUiOverlayStyle(
      statusBarColor: Colors.transparent,
      systemNavigationBarColor: theme.scaffoldBackgroundColor,
      systemNavigationBarDividerColor:
          theme.brightness == Brightness.light ? Colors.black : Colors.white,
      systemNavigationBarIconBrightness: theme.brightness == Brightness.light
          ? Brightness.dark
          : Brightness.light,
    ),
  );
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Simple Design',
      color: theme.scaffoldBackgroundColor,
      debugShowCheckedModeBanner: false,
      theme: theme,
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage>
    with SingleTickerProviderStateMixin {
  double sliderValue = 0.3;

  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return DefaultTabController(
      length: 3,
      initialIndex: 1,
      child: Scaffold(
        body: CustomScrollView(
          slivers: <Widget>[
            SDSliverAppBar(
                actions: <Widget>[
                  IconButton(
                    icon: Icon(Icons.star),
                    onPressed: () {},
                  )
                ],
                pinned: false,
                snap: true,
                floating: true,
                bottom: TabBar(tabs: [
                  Tab(
                    icon: Icon(Icons.slideshow),
                    text: "Slideshow",
                  ),
                  Tab(
                    icon: Icon(Icons.favorite_border),
                    text: "Favorites",
                  ),
                  Tab(
                    icon: Icon(Icons.music_note),
                    text: "Music",
                  ),
                ]),
                title: RichText(
                    text: TextSpan(
                        text: "Simple ",
                        style: Theme.of(context).textTheme.title,
                        children: [
                      TextSpan(
                          text: "Design",
                          style: Theme.of(context)
                              .textTheme
                              .title
                              .copyWith(fontWeight: FontWeight.normal))
                    ]))),
            SliverList(
                delegate: SliverChildListDelegate([
              SDDivider(
                height: 12.0,
              ),
              SDSectionHeader("Dialog and Button"),
              ListTile(
                leading: Icon(Icons.call_to_action),
                trailing: Row(
                  mainAxisSize: MainAxisSize.min,
                  children: <Widget>[
                    FlatButton(onPressed: () {}, child: Text("Flat Button")),
                    SizedBox(
                      width: 8.0,
                    ),
                    FlatButton(
                      color: Theme.of(context).primaryColor,
                      child: Text("Show Dialog"),
                      onPressed: () => showDialog(
                          context: context,
                          builder: (context) => SDDialog(
                                actionButton: SDActionButton(
                                  title: "Get Started",
                                  onPressed: () => Navigator.of(context).pop(),
                                ),
                                dismissButton: SDDismissButton(
                                  onPressed: () => Navigator.of(context).pop(),
                                  title: "Dismiss",
                                ),
                                title: "Welcome to Simple Design!",
                                content: Text(
                                  "With Simple Design we try to make the designing "
                                  "of your apps as simple, yet beautiful as possible, "
                                  "so you can focus on the important aspects of your app!",
                                ),
                              )),
                    ),
                  ],
                ),
              ),
              SDDivider(),
              SDSectionHeader("TextField"),
              ListTile(
                leading: Icon(Icons.text_fields),
                title: TextField(
                  decoration: InputDecoration(hintText: "Put in your username"),
                ),
              ),
              SDDivider(),
              SDSectionHeader("Slider"),
              ListTile(
                leading: Icon(Icons.adjust),
                title: Slider(
                    value: sliderValue,
                    onChanged: (n) {
                      setState(() {
                        sliderValue = n;
                      });
                    }),
              ),
              ListTile(
                leading: Icon(Icons.vertical_align_center),
                title: Slider(
                    divisions: 5,
                    label: (sliderValue * 10).toStringAsFixed(0),
                    value: sliderValue,
                    onChanged: (n) {
                      setState(() {
                        sliderValue = n;
                      });
                    }),
              ),
              ListTile(
                leading: Icon(Icons.location_disabled),
                title: Slider(
                  divisions: 5,
                  value: sliderValue,
                  onChanged: null,
                ),
              ),
              SDDivider(),
              SDSectionHeader("Cards"),
              Padding(
                padding: const EdgeInsets.all(12.0),
                child: SDCard(
                  mediaContent: Image.asset(
                    "assets/flower.jpg",
                    fit: BoxFit.cover,
                  ),
                  title: "Our Beautiful Planet",
                  subtitle: "by Manfred Richter",
                  content: Text(
                      "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, "
                      "sed diam nonumy eirmod tempor invidunt ut labore et dolore"
                      " magna aliquyam erat, sed diam voluptua. At"),
                  actions: <Widget>[
                    FlatButton(onPressed: () {}, child: Text("Read")),
                    IconButton(
                      icon: Icon(Icons.star_border),
                      color: theme.brightness == Brightness.light
                          ? Theme.of(context).primaryColor
                          : Theme.of(context).iconTheme.color,
                      onPressed: () {},
                    ),
                    IconButton(
                      icon: Icon(Icons.share),
                      color: theme.brightness == Brightness.light
                          ? Theme.of(context).primaryColor
                          : Theme.of(context).iconTheme.color,
                      onPressed: () {},
                    )
                  ],
                ),
              )
            ]))
          ],
        ),
      ),
    );
  }
}
